generics/services/survey.js

```diff
@@ -5,757 +5,873 @@
  * Description : Assessment service related information.
  */
 
- //dependencies
-const request = require('request');
+//dependencies
+const request = require('request')
 
-const ASSESSMENT_URL = process.env.ML_SURVEY_SERVICE_URL;
+const SURVEY_SERVICE_URL = process.env.SURVEY_SERVICE_URL
+/**
+ * Assessment create
+ * @function
+ * @name createAssessmentSolutionFromTemplate
+ * @param {String} token - logged in user token.
+ * @param {String} templateId - template id.
+ * @param {Object} bodyData - Body data
+ * @returns {JSON} - Create assessment from template solution.
+ */
+
+const createAssessmentSolutionFromTemplate = function (token, templateId, bodyData) {
+	return new Promise(async (resolve, reject) => {
+		try {
+			let assessmentCreateUrl =
+				SURVEY_SERVICE_URL + CONSTANTS.endpoints.ASSESSMENTS_CREATE + '?solutionId=' + templateId
+
+			const options = {
+				headers: {
+					'content-type': 'application/json',
+					'internal-access-token': process.env.INTERNAL_ACCESS_TOKEN,
+					'x-authenticated-user-token': token,
+				},
+				json: bodyData,
+			}
+
+			request.post(assessmentCreateUrl, options, assessmentCallback)
+
+			function assessmentCallback(err, data) {
+				let result = {
+					success: true,
+				}
+
+				if (err) {
+					result.success = false
+				} else {
+					let response = data.body
+					if (response.status === HTTP_STATUS_CODE['ok'].status) {
+						result['data'] = response.result
+					} else {
+						result.success = false
+					}
+				}
+				return resolve(result)
+			}
+		} catch (error) {
+			return reject(error)
+		}
+	})
+}
+
+/**
+ * Create Observation
+ * @function
+ * @name createObservationFromSolutionTemplate
+ * @param {String} solutionId - template id.
+ * @param {Object} bodyData - Body data
+ * @param {String} token - logged in user token.
+ * @param {String} programId -programId
+ * @param {Boolean} isExternalProgram - query external db for programDetails
+ * @returns {JSON} - Create assessment from template solution.
+ */
+
+const createObservationFromSolutionTemplate = function (solutionId, bodyData, token, programId, isExternalProgram) {
+	return new Promise(async (resolve, reject) => {
+		try {
+			let observationCreateUrl =
+				SURVEY_SERVICE_URL +
+				CONSTANTS.endpoints.CREATE_OBSERVATIONS +
+				'?solutionId=' +
+				solutionId +
+				'&programId=' +
+				programId +
+				'&isExternalProgram=' +
```

Comment:
@praveenKDass why do we need this key? Survey service should know if the solution has external program or not.